typed-protocols/src/Network/TypedProtocol/Driver.hs

```diff
@@ -0,0 +1,162 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE TypeInType #-}
+
+--TODO: just temporary while hacking:
```

Comment:
I think the comments in this module can be removed / update, we don't have `Duplex` here

Reply:
Agreed. I've not reviewed the comments here yet. I can delete the stale ones for this PR at least.